bin/fetch-engine.sh

```diff
@@ -0,0 +1,37 @@
+#! /usr/bin/env sh
+
+if [ $# -eq 1 ]
+then engine_name="$1"
+elif [ $# -eq 0 ]
+then
+  os=$(uname -s)
+  case "$os" in
+    Darwin)   os=darwin  ;;
+    Linux)    os=linux   ;;
+    FreeBSD)  os=freebsd ;;
+    OpenBSD)  os=openbsd ;;
+    NetBSD)   os=netbsd  ;;
+    CYGWIN*)  os=windows ;;
+    MINGW*)   os=windows ;;
+    MSYS*)    os=windows ;;
+    WINDOWS*) os=windows ;;
+    *)        echo "unhandled os: $os" ; exit 1 ;;
+  esac
+  arch=$(uname -m)
+  case $arch in
+    x86_64)   arch=amd64 ;;
+    i[36]86)  arch=386   ;;
+    aarch64)  arch=arm   ;;
+    arm*)     arch=arm   ;;
+    *)        echo "unhandled architecture: $arch" ; exit 1 ;;
+  esac
+  engine_name=file_engine_${os}_${arch}
+else
+  echo Error: fetch-engine.sh accepts at most 1 argument
+  exit 1
+fi
+
+engine_fpath=$(dirname "$0")/${engine_name}
+curl -fSsLo "$engine_fpath" \
```

Comment:
A good set of parameters to `curl`.
Preferably use their expanded synonyms; I think they stand for

```sh
curl --fail --show-error --remote-name --location
```

Reply:
Using the long option names make sense, for people who want to inspect the script.

Though `--remote-name` which is long for `-O`, isn't actually what should be used here, b/c it then the downloaded file would contain a suffix of `?raw=true`